chalice/awsclient.py

```diff
@@ -102,18 +122,60 @@ def _call_client_method_with_retries(self, method, kwargs):
         while True:
             try:
                 response = method(**kwargs)
-            except client.exceptions.InvalidParameterValueException:
+            except client.exceptions.InvalidParameterValueException as e:
                 # We're assuming that if we receive an
                 # InvalidParameterValueException, it's because
                 # the role we just created can't be used by
                 # Lambda so retry until it can be.
                 self._sleep(self.DELAY_TIME)
                 attempts += 1
-                if attempts >= self.LAMBDA_CREATE_ATTEMPTS:
+                message = e.response['Error'].get('Message', '')
+                retryable_message = (
+                    'The role defined for the function cannot be assumed '
```

Comment:
I am planning to just check to see if the string 'role' is included in the error message if that is fine?